### ISSUE TITLE
v2 VERP - Wrong "Sender" header when sending template tests

### ISSUE THREAD

**Issue by priethor** [Mailtrain-org/mailtrain#541]
Steps to reproduce:
1. Enable VERP
2. Edit a template content (not a campaign)
3. Send a test email with the template.

As VERP is enabled but we are not sending in a campaign context, the sender header in the resulting mail looks like this:

> Sender: [CAMPAIGN_ID].AWTsytRph.F-t196jPQ@xxx.yyy.zzz

The `[CAMPAIGN_ID]`  "merge tag" can't be resolved and the header has wrong characters for an email address, which makes email providers such as Gmail and Outlook reject the email.

**Comment by bures** [comment]
Good point. I didn't realize this interaction. I'll look into it if it is not very urgent.

**Comment by priethor** [comment]
It isn't, it can be solved by using an alternative non-VERP send configuration, as VERP and bounce handling doesn't make sense in the context of testing your own campaigns.

**Comment by bures** [comment]
That's a good point. Do you think the best way to avoid this would be just to disable VERP the emails sent for the template editor? 

I guess we have the same problem with VERP when dealing with transactional messages (i.e. subscription, etc.). Right?

**Comment by priethor** [comment]
You are right, transactional messages don't have a campaign context and will probably be affected, too. I haven't had a deep look at forms and transactional messages yet, I'll do it tomorrow or Thursday at most, but now I'm afraid if might have deeper implications.

For instance, as send configurations are campaign-based and not related to lists, which one is being used for transactional messages? I can imagine the Root one, so that means the "from" will be platform dependant and the same for all lists, no real multitenancy. Also, IMHO bounce handling can make sense in a subscription context, as if the confirmation email bounces it might not be a good idea to add the subscriber to the list, although this is probable not a big deal since:
- With a double opt in setup, the user won't confirm the subscription and therefore won't be fully subscribed.
- Upon next campaign send, it will bounce anyway.

**TL;DR;** Disabling VERP for template/campaign testing and transactional messages seems the best solution that doesn't require further changes.

**Comment by priethor** [comment]
I just noticed that this `[CAMPAIGN_ID]` not being resolved issue not only affects VERP, but every email sent out a campaign context. There are other headers affected, such as:
- **Return path** (only with VERP enabled)
- **X-FBL**: [CAMPAIGN_ID].r1NMEQeyX.HJ-Q4mlJQ
- **X-Msys-Api**: {"campaign_id":"[CAMPAIGN_ID].r1NMEQeyX.HJ-Q4mlJQ"}
- **X-SMTPAPI:** {"unique_args":{"campaign_id":"[CAMPAIGN_ID].r1NMEQeyX.HJ-Q4mlJQ"}}
- **X-Mailgun-Variables**: {"campaign_id":"[CAMPAIGN_ID].r1NMEQeyX.HJ-Q4mlJQ"}

**Comment by priethor** [comment]
Fixed in 30b361290b92c09d7c258345a5042faf5e2a015a.
